domaindriven-core/src/DomainDriven/Persistance/ForgetfulInMemory.hs

```diff
@@ -32,12 +34,15 @@ data ForgetfulInMemory model event = ForgetfulInMemory
 instance ReadModel (ForgetfulInMemory model e) where
     type Model (ForgetfulInMemory model e) = model
     type Event (ForgetfulInMemory model e) = e
-    applyEvent ff = apply ff
+    applyEvent = apply
+    getModel :: ForgetfulInMemory model e -> IO (Model (ForgetfulInMemory model e))
     getModel ff = readIORef $ stateRef ff
     getEventList ff = readIORef $ events ff
-    getEventStream ff = do
-        l <- liftIO $ getEventList ff
-        S.fromList l
+    getEventStream ff =
+        Stream.bracketIO
+            (getEventList ff)
+            (const (pure ()))
+            Stream.fromList
```

Comment:
Shouldn't this just be [fromEffect](https://hackage.haskell.org/package/streamly-core-0.2.2/docs/Streamly-Data-Stream.html#v:fromEffect)?

As in, `fromEffect (getEventList ff)`

Reply:
Ok nvm, you are right.